staging/src/k8s.io/kubectl/pkg/cmd/label/label.go

```diff
@@ -140,76 +186,92 @@ func NewCmdLabel(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobr
-	usage := "identifying the resource to update the labels"
-	cmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
-	cmdutil.AddDryRunFlag(cmd)
-	cmdutil.AddFieldManagerFlagVar(cmd, &o.fieldManager, "kubectl-label")
-	cmdutil.AddLabelSelectorFlagVar(cmd, &o.selector)
+	flags.AddFlags(cmd)
 
 	return cmd
 }
 
-// Complete adapts from the command line args and factory to the data required.
-func (o *LabelOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
+// ToOptions converts from CLI inputs to runtime inputs
+func (f *LabelFlags) ToOptions(cmd *cobra.Command, args []string) (*LabelOptions, error) {
 	var err error
 
-	o.RecordFlags.Complete(cmd)
-	o.Recorder, err = o.RecordFlags.ToRecorder()
+	f.RecordFlags.Complete(cmd)
+	f.Recorder, err = f.RecordFlags.ToRecorder()
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.outputFormat = cmdutil.GetFlagString(cmd, "output")
-	o.dryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	f.OutputFormat = cmdutil.GetFlagString(cmd, "output")
+	f.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	o.ToPrinter = func(operation string) (printers.ResourcePrinter, error) {
-		o.PrintFlags.NamePrintFlags.Operation = operation
+	toPrinter := func(operation string) (printers.ResourcePrinter, error) {
+		f.PrintFlags.NamePrintFlags.Operation = operation
 		// PrintFlagsWithDryRunStrategy must be done after NamePrintFlags.Operation is set
-		cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.dryRunStrategy)
-		return o.PrintFlags.ToPrinter()
+		cmdutil.PrintFlagsWithDryRunStrategy(f.PrintFlags, f.DryRunStrategy)
+		return f.PrintFlags.ToPrinter()
 	}
 
 	resources, labelArgs, err := cmdutil.GetResourcesAndPairs(args, "label")
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.resources = resources
-	o.newLabels, o.removeLabels, err = parseLabels(labelArgs)
+
+	newLabels, removeLabels, err := parseLabels(labelArgs)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	if o.list && len(o.outputFormat) > 0 {
-		return fmt.Errorf("--list and --output may not be specified together")
+	if f.List && len(f.OutputFormat) > 0 {
+		return nil, fmt.Errorf("--list and --output may not be specified together")
 	}
 
-	o.namespace, o.enforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	namespace, enforceNamespace, err := f.Factory.ToRawKubeConfigLoader().Namespace()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	o.builder = f.NewBuilder()
-	o.unstructuredClientForMapping = f.UnstructuredClientForMapping
+	builder := f.Factory.NewBuilder()
+	unstructuredClientForMapping := f.Factory.UnstructuredClientForMapping
 
-	return nil
+	return &LabelOptions{
+		FilenameOptions: f.FilenameOptions,
+
+		PrintFlags: f.PrintFlags,
```

Comment:
We already have `ToPrinter`, why we are also storing `PrintFlags` in options?

Reply:
Hmm, your review made me realize the difference between Flags and Options. Did it, thanks!